src/gwc/src/main/java/org/geoserver/gwc/layer/DefaultTileLayerCatalog.java

```diff
@@ -292,7 +293,7 @@ private GeoServerTileLayerInfoImpl depersist(final Resource res) throws IOExcept
             LOGGER.fine("Depersisting GeoServerTileLayerInfo from " + res.path());
         }
         GeoServerTileLayerInfoImpl info;
-        try(Reader reader = new InputStreamReader(res.in(), "UTF-8")) {
+        try(Reader reader = new InputStreamReader(new ByteArrayInputStream(res.getContents()), "UTF-8")) {
```

Comment:
See below comment on res.getContents()....

I more had in mind implementing FileSystemResourceStore implementation of in() to use your more efficient technique:

````
public InputStream in() {
   ...
   return new ByteArrayInputStream( java.nio.file.Files.readAllBytes(file.toPath())){
      public void close() throws IOException {..}
      protected void finalize() throws IOException {..}
   }
}
````

So it is harder to dodge the most efficient technique.